aplib.net-demo/Assets/Scripts/Configuration.cs

```diff
@@ -0,0 +1,277 @@
+        else if (GridSizeX < MinGridSize) GridSizeX = MinGridSize;
+
+        if (GridSizeZ > MaxGridSize) GridSizeZ = MaxGridSize;
+        else if (GridSizeZ < MinGridSize) GridSizeZ = MinGridSize;
+
+        // Update value.
+        GridVolume = GridSizeX * GridSizeZ;
+
+        // Check room value.
+        if (RoomAmount > GridVolume * _maxPercentage) RoomAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (RoomAmount < 0) RoomAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check ItemAmmoAmount value.
+        if (ItemAmmoAmount > GridVolume * _maxPercentage) ItemAmmoAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (ItemAmmoAmount < 0) ItemAmmoAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check ItemHealthAmount value.
+        if (ItemHealthAmount > GridVolume * _maxPercentage) ItemHealthAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (ItemHealthAmount < 0) ItemHealthAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check ItemRageAmount value.
+        if (ItemRageAmount > GridVolume * _maxPercentage) ItemRageAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (ItemRageAmount < 0) ItemRageAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check EnemyMeleeAmount value.
+        if (EnemyMeleeAmount > GridVolume * _maxPercentage) EnemyMeleeAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (EnemyMeleeAmount < 0) EnemyMeleeAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check EnemyRangedAmount value.
+        if (EnemyRangedAmount > GridVolume * _maxPercentage) EnemyRangedAmount = Mathf.CeilToInt(GridVolume * _maxPercentage);
+        else if (EnemyRangedAmount < 0) EnemyRangedAmount = Mathf.CeilToInt(GridVolume * _minPercentage);
+
+        // Check if items and enemies don't go over grid vol * _maxPrecentage(0.7f)
+        CheckItemOverflow();
+        CheckEnemyOverflow();
+
+        // Print correct number in text field. Skip this when testing.
+        if (!_test)
+        {
+            inputFieldGridW.text = GridSizeX.ToString();
+            inputFieldGridH.text = GridSizeZ.ToString();
+            inputFieldRoom.text = RoomAmount.ToString();
+            inputFieldItemAmmo.text = ItemAmmoAmount.ToString();
+            inputFieldItemHealth.text = ItemHealthAmount.ToString();
+            inputFieldItemRage.text = ItemRageAmount.ToString();
+            inputFieldEnemyMelee.text = EnemyMeleeAmount.ToString();
+            inputFieldEnemyRanged.text = EnemyRangedAmount.ToString();
+
+            SetValues();
+        }
+    }
+
+    public void CheckItemOverflow()
+    {
+        TotalItemCount = ItemAmmoAmount + ItemHealthAmount + ItemRageAmount;
+        if (TotalItemCount > GridVolume * _maxPercentage)
+        {
+            int maxCount = Mathf.CeilToInt(GridVolume * _maxPercentage / 3);
```

Comment:
Define this magic number somewhere at the top, give it a meaningful name and summary and maybe even SerliazeField it

Reply:
TotalItemCount? That changes everytime when the grid size is changed. So really defining this is not an option, same with maxCount

Reply:
I did give TotalItemCount a better description.